DIPs/DIP1012.md

```diff
@@ -96,86 +116,71 @@ Encompassed:
+Use of the current attributes that are not prefixed by an `@`, such as `pure` and `nothrow`, and optionally other modifiers that are attribute-like such as `final`, will be changed to refer to the `core.attribute` symbols, and thus their use without the leading `@` will be deprecated.
+
+During the deprecation phase where `pure` and `nothrow` appear that affect the attributes of a symbol (i.e. their current usage, as opposed to referring to the druntime symbol for e.g. a template parameter) the compiler shall treat `pure` and `nothrow` (without a leading `@`) as though they did have a leading `@`.
 
-No other breaking changes are intended, although the introduction of the new `enum` symbols to be implicitly imported by `object.d` may break some code if the chosen names clash (unlikely).
+No other breaking changes are intended, although the introduction of the new `enum` symbols to be implicitly imported by `object.d` may break some code if the chosen names clash (unlikely) but can be mitigated with `@future` if need be.
 
 ### Examples
 
 `module foo;` 
 
 will become implicitly 
 
-`@core.attribute.defaultAttributeSet module foo;` 
-
-with respect to attributes (`core.attribute` will be implicitly imported, by `object.d`), if no attributes from `core.attribute` are attached.
-
-Attribute groups may be selectivly added to the module declaration, so that:
- 
-`@nogc module foo;` 
- 
-means that all symbols in this module are implicity `@nogc` (with `nogc` referring to `core.attribute.GarbageCollectedness.nogc` via an alias in `core.attribute`),
- but otherwise has all the same defaults as the default attribute set.
- 
-`@nogc @core.attribute.GarbageCollectedness.gc module foo;` 
-shall be an error because there are two explicit mutually exclusive attributes.
-Likewise 
 ```d
 module foo;
-@nogc @core.attribute.GarbageCollectedness.gc 
-void bar() { };
-```
-shall be an error for the same reasons.
+@core.attribute.defaultAttributeSet:
+``` 
+
+Modules with out a module declaration also have `defaultAttributeSet` applied to all symbols.
  
 ```d
-@nogc module foo;
- 
-// bar overrides the default @nogc'ness of the module and is @gc
-@core.attribute.GarbageCollectedness.gc void bar() {auto a = new int;} 
+module foo;
+// overrides `core.attribute.defaultAttributeSet`'s GarbageCollectedness attribute it is
+// applied after defaultAttributeSet and therefore beats it by "last applied wins" rule
+@nogc:
+
+// bar overrides the default @nogc'ness above and is @gc
+// `gc` is an alias for `core.attribute.GarbageCollectedness.gc`
+@gc void bar() {auto a = new int;} 
  
 // baz's gc'ness is determined by someOtherFunction
-@core.attribute.GarbageCollectedness.inferred void baz() { someOtherFunction(); }
+@infer!(nogc) void baz() { someOtherFunction(); }
```

Comment:
What's the point of explicit inference? The compiler currently infers as many covariant attributes as possible. Functions at ABI boundary (export) will never support inference due to separate compilation.

Reply:
Ping on this question.